### PR TITLE
SEO GB Extension: Add

### LINKED ISSUE #27692: Gutenberg: SEO Description missing from editor sidebar
#### Steps to reproduce
1. Starting at URL: https://wpcalypso.wordpress.com/gutenberg/
2. Start a new post

#### What I expected
I expected to see the sidebar settings matching the existing settings

#### What happened instead
In the sidebar the SEO Description section is missing

#### Browser / OS version
All

#### Screenshot / Video
image

### PR DESCRIPTION
**Pull request by ockham**
Requires https://github.com/Automattic/jetpack/pull/11333

#### Changes proposed in this Pull Request

Add a Gutenberg extension to include an SEO description panel in the sidebar.

image

For comparison, this is what the legacy SEO panel in the Calypso Classic Editor looks like:

image

#### Testing instructions

* Build blocks:
  ```
  npm ci
  npx lerna bootstrap --concurrency=2 --scope='@automattic/jetpack-blocks'
  npx lerna run build --stream --scope='@automattic/jetpack-blocks' -- -- --output-path /path/to/jetpack/_inc/blocks/
  ```
   Remember to change output path to correct.
* Apply this branch and Jetpack's `add/seo-post-meta` branch to your local environment
* Open the Jetpack sidebar or hit "publish" on new post/page
* find the 'SEO Description' panel, enter some text into the textfield.
* Save draft/publish
* Verify that the SEO string is persisted.

Fixes #27692

### CODE REVIEW

**Comment by ockham** [comment]
I tested setting the SEO desc string from the classic Calypso editor and verified it's being persisted there -- but that string doesn't show up in the Gutenberg extension panel, so apparently we're not even fetching it from the endpoint.

A closer look at `gutenbergState()['core/editor'].currentPost` confirms that we're not getting a lot of metadata:

image

Maybe we need to explicitly include the meta field in the (core) `wp/v2` post endpoint?

**Comment by ockham** [comment]
> Maybe we need to explicitly include the meta field in the (core) `wp/v2` post endpoint?

I think it's really that -- we're just missing a `register_meta()` call on the server side. (All the other registrations in the codebase are for wpcom or JP endpoints.)

**Comment by ockham** [comment]
> > Maybe we need to explicitly include the meta field in the (core) `wp/v2` post endpoint?
> 
> I think it's really that -- we're just missing a `register_meta()` call on the server side. (All the other registrations in the codebase are for wpcom or JP endpoints.)

Implemented in https://github.com/Automattic/jetpack/pull/11323

File: client/gutenberg/extensions/seo/editor.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+// @TODO: Replace with Gutenberg variables
+$dark-gray-500: #555d66;
```
Comment (simison) [inline client/gutenberg/extensions/seo/editor.scss]: Not used in this file?

File: client/gutenberg/extensions/seo/editor.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+.jetpack-seo-message-box {
+	background-color: #edeff0;
```
Comment (simison) [inline client/gutenberg/extensions/seo/editor.scss]: Was this supposed to be that `$dark-gray-500`?

File: client/gutenberg/extensions/seo/panel.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/** @format */
```
Comment (simison) [inline client/gutenberg/extensions/seo/panel.jsx]: We can remove these, Prettier now does all the files. 🙌

Comment (ockham) [reply client/gutenberg/extensions/seo/panel.jsx]: Ugh, leftover. Thanks, good spot!

File: client/gutenberg/extensions/seo/editor.scss (hunk 1 above)
Comment (ockham) [reply client/gutenberg/extensions/seo/editor.scss]: Copypasta :grimacing:

File: client/gutenberg/extensions/seo/editor.scss (hunk 2 above)
Comment (ockham) [reply client/gutenberg/extensions/seo/editor.scss]: Nah, also copypasta from Publicize. This corresponds to `$light-gray-300`, so I've changed it to that for now.

**Comment by simison** [comment]
Did kinda blind rebase, I'll have another look later if this still works.

**Comment by simison** [comment]
Noting that this won't be available at sites before https://github.com/Automattic/jetpack/pull/11333 is merged.

**Pull request merged**